Gems/Atom/RHI/Vulkan/Code/Source/RHI/Device.cpp

```diff
@@ -829,6 +847,95 @@ namespace AZ
             m_releaseQueue.Notify(notifyFunction);
         }
 
+        RHI::ShadingRateImageValue Device::ConvertShadingRate(RHI::ShadingRate rate)
+        {
+            ShadingRateImageMode mode = GetImageShadingRageMode();
+            if (mode == ShadingRateImageMode::ImageAttachment)
+            {
+                // Fragment sizes are encoded in a single texel as follows:
+                // size(w) = 2^((texel/4) & 3)
+                // size(h) = 2^(texel & 3)
+
+                uint8_t rate_w, rate_h;
+                switch (rate)
+                {
+                case RHI::ShadingRate::Rate1x1:
+                    rate_w = rate_h = 0;
```

Comment:
Yea, I know the encoding uses a mask as you pointed out, I just meant that in code, I could see a person being confused if they looked at `rate_w` and it was `0` (since this is not actually the rate). I'd suggest a name like `encoded_rate_w` maybe? That said, not a huge deal at all